configure.ac

```diff
@@ -88,22 +109,32 @@ else
    AC_DEFINE([FORTRANUNDERSCORE], [1], [use single underscore for Fortran wrappers])
 fi
 
+AC_MSG_CHECKING([whether PAPI is to be enabled])
+AC_ARG_ENABLE([papi], [AS_HELP_STRING([--enable-papi],
+              [enable papi])])
+test "x$enable_papi" = xyes || enable_papi=no
+AC_MSG_RESULT($enable_papi)
+
```

Comment:
I usually do all my ARG processing (enables/disables) at the top of the script, the all my $enable_whatever variables are set to "yes" or "no" and the rest of the script I can use them to make decisions.